### PR TITLE
Feature/add full text search for articles

### PR DESCRIPTION
**Pull request by Andriy1328**
1. Add full text search for articles.
2. Create design for search field.
Screenshot from 2022-08-18 16-59-40

### CODE REVIEW

File: app/views/articles/index.html.erb [hunk 1]
```diff
@@ -7,20 +7,32 @@
...
+              <span></span>
+            <% end %>
+          </div>    
+        </div>
+      <% end %>
+    </div>   
   <div class='articles-tags'>
     <ul class='articles-list'>
-      <% @articles.each do |article| %>
+     <% ( !params[:query] || params[:query].empty? ? @articles : @search ).each do |article| %>
```
Comment (mmrshk) [inline app/views/articles/index.html.erb:25]: this logic should be in controller

File: app/views/shared/_article.html.erb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <div><%= link_to article.title, article_path(article), class: 'article-title' %></div>
-<div class='ck-content'><%= sanitize truncate(article.text, escape: false, length: 600,) { link_to 'Continue', article_path(article) } %></div>
+<div class='ck-content'><%= sanitize truncate((article.text), escape: false, length: 600,) { link_to 'Continue', article_path(article) } %></div>
```
Comment (mmrshk) [inline app/views/shared/_article.html.erb:2]: why you need this change?

File: config/routes.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
     resources :tags, only: %i[index show]
     root to: "home#index"
   end
-
+
```
Comment (mmrshk) [inline config/routes.rb:28]: extra space

File: app/controllers/articles_controller.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@
 class ArticlesController < ApplicationController
   def index
     @articles = Article.published.in_language(extract_locale).sorted_desc
+    @search = Article.articles_search(params[:query])
```
Comment (mmrshk) [inline app/controllers/articles_controller.rb:6]:
```
@articles = if params[:query] do
else
end  
```

File: app/models/article.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class Article < ApplicationRecord
+  include PgSearch::Model
```
Comment (mmrshk) [inline app/models/article.rb:4]: add empty line

File: app/models/article.rb [hunk 6]
```diff
@@ -25,4 +26,20 @@ class Article < ApplicationRecord
...
+                      name: 'B'
+                    }   
+                  },
+                  using: {
+                    tsearch: {
+                      prefix: true,
+                      dictionary: "english",
+                    }
+                  }
+end
```
Comment (mmrshk) [inline app/models/article.rb:45]: EOL

File: config/routes.rb (hunk 3 above)
Comment (mmrshk) [inline config/routes.rb:28]: extra line, but without white space

**Review by mmrshk — APPROVED** [review]
LGTM! Fix spacing and conflicts.

File: app/models/article.rb (hunk 6 above)
Comment (Andriy1328) [reply app/models/article.rb]: Fixed

File: app/controllers/articles_controller.rb (hunk 4 above)
Comment (Andriy1328) [reply app/controllers/articles_controller.rb]: Fixed

File: app/views/shared/_article.html.erb (hunk 2 above)
Comment (Andriy1328) [reply app/views/shared/_article.html.erb]: Fixed

File: app/views/articles/index.html.erb (hunk 1 above)
Comment (Andriy1328) [reply app/views/articles/index.html.erb]: Fixed

File: config/routes.rb (hunk 3 above)
Comment (Andriy1328) [reply config/routes.rb]: Fixed

**Pull request merged**